features/0360-use-did-key/README.md

```diff
@@ -0,0 +1,106 @@
+
+Note that simply knowing the key type is not necessarily sufficient to be able to use the key. The cryptography supporting the processing data using the key must also be available in the agent. However, the multicodec and did:key capabilities will simplify adding support for new key types in the future.
+
+## Tutorial
+
+An example of the use of the replacement of a verkey with `did:key` can be found in the [~service decorator RFC](https://github.com/hyperledger/aries-rfcs/tree/master/features/0056-service-decorator). Notably in the example at the beginning of the [tutorial](https://github.com/hyperledger/aries-rfcs/tree/master/features/0056-service-decorator#tutorial) section, the verkeys in the `recipientKeys` and `routingKeys` items would be changed from native keys to use `did:key` as follows:
+
+``` jsonc
+{
+    "@type": "somemessagetype",
+    "~service": {
+        "recipientKeys": ["did:key:z6MkmjY8GnV5i9YTDtPETC2uUAW6ejw3nk5mXF5yci5ab7th"],
+        "routingKeys": ["did:key:z6MkmjY8GnV5i9YTDtPETC2uUAW6ejw3nk5mXF5yci5ab7th"]
+        "serviceEndpoint": "https://example.com/endpoint"
+    }
+}
+```
+
+Thus, `8HH5gYEeNc3z7PYXmd54d4x6qAfCNrqQqEB3nS7Zfu7K` becomes `did:key:z6MkmjY8GnV5i9YTDtPETC2uUAW6ejw3nk5mXF5yci5ab7th` using the following transformations:
+
+- Start with the original (presumably) base58 ed25519 public key
+  - Since we have only a naked public key, we must assume the format
+- Base58 decode the public key to get the hex version
+- Multicodec: Prefix with the algorithm type. In this case, `ED01` as the key type is ed25519.
+- Multibase: Base58 encode the result and prefix that with "z" to indicate base58 encoding
+- DID: Prefix that with the DID Method prefix `did:key:`
+
+The transformation above is **only for illustration within this RFC**. The [`did:key`](https://digitalbazaar.github.io/did-method-key) specification is the definitive source for the appropriate transformations.
+
+The `did:key` method uses the strings that are the DID, public key and key type to construct ("resolve") a DIDDoc based on a template defined by the `did:key` specification. Further, the `did:key` resolver generates, in the case of an ed25519 public signing key, a key that can be used as part of a Diffie-Hellman exchange appropriate for encryption in the `keyAgreement` section of the DIDDoc. Presumably, as `did:key` supports other key types, similar DIDDoc templates will become part of the specification. Key types that don't support a signing/key exchange transformation would not have a `keyAgreement` entry in the resolved DIDDoc.
+
+The following currently implemented RFCs would be affected by adoption of this RFC. In these RFCs, the JSON items that currently contain naked public keys (mostly the items `recipientKeys` and `routingKeys`) would be changed to use `did:key` references where applicable. Note that in these items public DIDs could also be used if applicable for a given use case.
```

Comment:
I think we should separate the discussion of "naked" keys in the envelope from the other "naked" keys topics.

In particular, I think the JWE envelope ([RFC 360](https://github.com/hyperledger/aries-rfcs/tree/master/features/0334-jwe-envelope)) should continue to support (and use) JWKs. 
- A JWK representation is well-known (and implemented) for JWEs: https://tools.ietf.org/html/rfc7516#section-4.1.5. 
- The underlying spec for ECDH-1PU references JWK for the "epk" field: https://tools.ietf.org/html/rfc7518#section-4.6.1.1

Reply:
@troyronda - Removed encryption envelope from list of "to be changed" protocols; added to do about discussing the use or not of did:key with the encryption envelope.